logstash-core/lib/logstash/api/commands/node.rb

```diff
@@ -0,0 +1,117 @@
+require "logstash/api/commands/base"
+
+module LogStash
+  module Api
+    module Commands
+      class Node < Commands::Base
+        def all
+          {
+            :pipeline => pipeline,
+            :os => os,
+            :jvm => jvm,
+            :hot_threads => hot_threads
+          }           
+        end
+        
+        def pipeline
+          extract_metrics(
+            [:stats, :pipelines, :main, :config],
+            :workers, :batch_size, :batch_delay
+          )
+        end
+
+        def os
+          {
+            :name => java.lang.System.getProperty("os.name"),
+            :arch => java.lang.System.getProperty("os.arch"),
+            :version => java.lang.System.getProperty("os.version"),
+            :available_processors => java.lang.Runtime.getRuntime().availableProcessors()
+          }
+        end
+
+        def jvm
+          memory_bean = ManagementFactory.getMemoryMXBean()
+          {
+            :pid =>  ManagementFactory.getRuntimeMXBean().getName().split("@").first.to_i,
+            :version => java.lang.System.getProperty("java.version"),
+            :vm_name => java.lang.System.getProperty("java.vm.name"),
+            :vm_version => java.lang.System.getProperty("java.version"),
+            :vm_vendor => java.lang.System.getProperty("java.vendor"),
+            :vm_name => java.lang.System.getProperty("java.vm.name"),            
+            :start_time_in_millis => started_at,
+            :mem => {
+              :heap_init_in_bytes => (memory_bean.getHeapMemoryUsage().getInit() < 0 ? 0 : memory_bean.getHeapMemoryUsage().getInit()),
```

Comment:
I agree that it isn't so great to read, but I think its better than splitting into a separate line where it takes up less horizontal but more vertical space.

I'm basically stealing this from https://github.com/elastic/elasticsearch/blob/34da01f794ba892176376aafc18e901cacdd0fce/core/src/main/java/org/elasticsearch/monitor/jvm/JvmInfo.java#L72

I feel like the ES team probably made the same tradeoff